src/ekat/ekat_session.cpp

```diff
@@ -32,6 +32,18 @@ void initialize_kokkos () {
 # elif defined KOKKOS_ENABLE_HIP
   const auto ret = hipGetDeviceCount(&nd);
   const bool ok = ret == hipSuccess;
+# elif defined KOKKOS_ENABLE_SYCL
+  nd = 0;
+  auto gpu_devs = sycl::device::get_devices(sycl::info::device_type::gpu);
```

Comment:
This part is flying above my head. I'm just going to assume that it does the SYCL-equivalent of what we do for CUDA and HIP?

Reply:
In functionality yes! :) 

Since SYCL almost mimics OpenCL, there is a bit more functionality on how devices are seen. For the `if` case, one can partition a given GPU into several pseudo sub-devices like CUDA's MIG mode. For the `else`, it is just standard, treat the device as it is. For some context, MI-250X is seen as 2 GPU by default on crusher though it is physically one.

For best performance on Intel-GPU, it is recommended to use the `if` case where the device gets partitioned based on the NUMA-domain. To be generic, if the device cant be partitioned it falls through the else case.